lib/active_merchant/billing/gateways/litle.rb

```diff
@@ -39,6 +39,88 @@ def purchase(money, payment_method, options = {})
+              doc.detailTax do
+                doc.taxIncludedInTotal(level_3_data[:tax_included_in_total]) if level_3_data[:tax_included_in_total]
+                doc.taxAmount(level_3_data[:tax_amount]) if level_3_data[:tax_amount]
+                doc.taxRate(level_3_data[:tax_rate]) if level_3_data[:tax_rate]
+                doc.taxTypeIdentifier(level_3_data[:tax_type_identifier]) if level_3_data[:tax_type_identifier]
+                doc.cardAcceptorTaxId(level_3_data[:card_acceptor_tax_id]) if level_3_data[:card_acceptor_tax_id]
+              end
+              add_line_item_information_for_level_3_visa(doc, payment_method, level_3_data)
+            when 'master'
+              doc.customerReference :customerReference, level_3_data[:customer_code] if level_3_data[:customer_code]
+              doc.salesTax(level_3_data[:total_tax_amount]) if level_3_data[:total_tax_amount]
+              doc.detailTax do
+                doc.taxIncludedInTotal(level_3_data[:tax_included_in_total]) if level_3_data[:tax_included_in_total]
+                doc.taxAmount(level_3_data[:tax_amount]) if level_3_data[:tax_amount]
+                doc.cardAcceptorTaxId :cardAcceptorTaxId, level_3_data[:card_acceptor_tax_id] if level_3_data[:card_acceptor_tax_id]
+              end
+              doc.lineItemData do
+                level_3_data[:line_items].each do |line_item|
+                  doc.itemDescription(line_item[:item_description]) if line_item[:item_description]
+                  doc.productCode(line_item[:product_code]) if line_item[:product_code]
+                  doc.quantity(line_item[:quantity]) if line_item[:quantity]
+                  doc.unitOfMeasure(line_item[:unit_of_measure]) if line_item[:unit_of_measure]
+                  doc.lineItemTotal(line_item[:line_item_total]) if line_item[:line_item_total]
+                end
+              end
+            end
+          end
+        end
+      end
+
+      def add_line_item_information_for_level_3_visa(doc, payment_method, level_3_data)
+        doc.lineItemData do
+          level_3_data[:line_items].each do |line_item|
+            doc.itemSequenceNumber(line_item[:item_sequence_number]) if line_item[:item_sequence_number]
+            doc.commodityCode(line_item[:commodity_code]) if line_item[:commodity_code]
+            doc.itemDescription(line_item[:item_description]) if line_item[:item_description]
+            doc.productCode(line_item[:product_code]) if line_item[:product_code]
+            doc.quantity(line_item[:quantity]) if line_item[:quantity]
+            doc.unitOfMeasure(line_item[:unit_of_measure]) if line_item[:unit_of_measure]
+            doc.taxAmount(line_item[:tax_amount]) if line_item[:tax_amount]
+            doc.itemDiscountAmount(line_item[:discount_per_line_item]) if line_item[:discount_per_line_item]
+            doc.unitCost(line_item[:unit_cost]) if line_item[:unit_cost]
```

Comment:
💬 
According the documentation you would need add a validation to avoid 0 values in unit Cost tag. and check the same for item Discount Amount.

Reply:
Corrected, please review.